internal/pkg/iptablesutils/iptables.go

```diff
@@ -61,8 +61,20 @@ func DetectIPTablesMode(k0sBinPath string) (string, error) {
 		logrus.Info("kube-related iptables entries not found, go with iptables-legacy")
 		return ModeLegacy, nil
 	}
-	logrus.Info("kube-related iptables entries not found, go with iptables-nft")
-	return ModeNFT, nil
+
+	iptablesPath, err := exec.LookPath("iptables")
+	if err != nil {
+		return "", err
```

Comment:
Do we want to err out here? I'd have expected that the err is ignored and `ModeLegacy, nil` gets returned.

Reply:
The idea here is that if there is no working `iptables` binary on the host, then it means we failed to detect the mode and 
`DetectIPTablesMode` return error. It means we don't know what the mode is on the host.

Reply:
> if there is no working iptables binary on the host, then it means we failed to detect the mode

that would mean you cannot run k0s at all on a host that does not have iptables bin on the host. Which is not what we want.

Reply:
> that would mean you cannot run k0s at all on a host that does not have iptables bin on the host. Which is not what we want.

Yes, but in case `DetectIPTablesMode` returns an error we stick to the defaults [here](https://github.com/k0sproject/k0s/pull/2277/files#diff-c4db305f5cd0d057a727228694bf6fd480aeb5dead2c8fff79cdd73f0c5c9326R99), so everything works as expected. 

But maybe we need to refactor it a bit to make the code easier to understand.

Reply:
> that would mean you cannot run k0s at all on a host that does not have iptables bin on the host. Which is not what we want.

No. It meant we fail to detect the host's iptables mode, in which case we fallback to kernel major version check and use nftables unless kernel is older than 5.x.y